lib/propolis/src/vmm/mem.rs

```diff
@@ -1059,6 +1065,61 @@ impl<T: Copy + FromBytes> Iterator for GuestData<MemMany<'_, T>> {
     }
 }
 
+/// A 52-bit physical page number, i.e., the ordinal index of a 4 KiB page of
+/// guest memory.
+#[derive(Clone, Copy, Ord, PartialOrd, Eq, PartialEq)]
+pub(crate) struct Pfn(u64);
+
+impl std::fmt::Debug for Pfn {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "Pfn({:x})", self.0)
```

Comment:
nitpicky: perhaps we should stick a 
```suggestion
        write!(f, "Pfn({:#x})", self.0)
```
in here to get a leading `0x` to make it clearer that it's hex?

Reply:
Fixed in 6e61e6e.